py/sentry_streams/runner.py

```diff
@@ -92,7 +98,7 @@ def main() -> None:
     }
 
     pipeline: Pipeline = pipeline_globals["pipeline"]
-    runtime = load_adapter(AdapterType.FLINK, environment_config)
+    runtime: FlinkAdapter = load_adapter(AdapterType.FLINK, environment_config)
```

Comment:
Why does this need to know about `FlinkAdapter` ? I'd assume you added the type hint here because `RuntimeTranslator(runtime)` somehow needs to know the specific Adapter. But that sounds coutner to the design that abstracts the adapter away from the translation code.